smithy-linters/src/main/java/software/amazon/smithy/linters/NoninclusiveTermsValidator.java

```diff
@@ -0,0 +1,224 @@
+
+/**
+ * <p>Validates that all shape names, and values does not contain non-inclusive terms.
+ * *
+ * <p>See AbstractModelTextValidator for scan implementation details.
+ */
+public final class NoninclusiveTermsValidator extends AbstractValidator {
+    static final Map<String, List<String>> BUILT_IN_NONINCLUSIVE_TERMS = MapUtils.of(
+            "master", ListUtils.of("primary", "parent", "main"),
+            "slave", ListUtils.of("secondary", "replica", "clone", "child"),
+            "blacklist", ListUtils.of("denylist"),
+            "whitelist", ListUtils.of("allowlist")
+        );
+
+    public static final class Provider extends ValidatorService.Provider {
+        public Provider() {
+            super(NoninclusiveTermsValidator.class, node -> {
+                NodeMapper mapper = new NodeMapper();
+                return new NoninclusiveTermsValidator(
+                        mapper.deserialize(node, NoninclusiveTermsValidator.Config.class));
+            });
+        }
+    }
+
+    /**
+     * NoninclusiveTermsValidator configuration.
+     */
+    public static final class Config {
+        private Map<String, List<String>> noninclusiveTerms = MapUtils.of();
+        private boolean appendDefaults = false;
+
+        public Map<String, List<String>> getNonInclusiveTerms() {
+            return noninclusiveTerms;
+        }
+
+        public void setNoninclusiveTerms(Map<String, List<String>> terms) {
+            this.noninclusiveTerms = terms;
+        }
+
+        public boolean isAppendDefaults() {
+            return appendDefaults;
+        }
+
+        public void setAppendDefaults(boolean appendDefaults) {
+            this.appendDefaults = appendDefaults;
+        }
+    }
+
+    final Map<String, List<String>> termsMap;
+
+    private NoninclusiveTermsValidator(Config config) {
+        if (config.isAppendDefaults()) {
+            termsMap = new HashMap<>(BUILT_IN_NONINCLUSIVE_TERMS);
+            termsMap.putAll(config.getNonInclusiveTerms());
+        } else if (!config.getNonInclusiveTerms().isEmpty()) {
+            termsMap = MapUtils.copyOf(config.getNonInclusiveTerms());
+        } else {
+            termsMap = new HashMap<>(BUILT_IN_NONINCLUSIVE_TERMS);
+        }
+    }
+
+    /**
+     * Runs a full text scan on a given model and stores the resulting TextOccurrences objects.
+     *
+     * Namespaces are checked against a global set per model.
+     *
+     * @param model Model to validate.
+     * @return a list of ValidationEvents found by the implementer of getValidationEvents per the
+     *          TextOccurrences provided by this traversal.
+     */
+    @Override
+    public List<ValidationEvent> validate(Model model) {
+        TextIndex textIndex = TextIndex.of(model);
+        List<ValidationEvent> validationEvents = new ArrayList<>();
+        for (TextInstance text : textIndex.getTextInstances()) {
+            getValidationEvents(text, validationEvent -> {
```

Comment:
Instead of using an anonymous consumer here, this should be refactored towards `validationEvents.addAll(getValidationEvents(text))`.

Reply:
Done